### ISSUE TITLE
Brave update makes extensions unloadable & profile deletion is impossible

### ISSUE THREAD

**Issue by snowbound** [brave/brave-browser#16034]
## Description 
Upgraded to Version 1.26.34 Chromium: 91.0.4472.57 (Official Build) beta (64-bit) brave does not load extensions

## Steps to Reproduce

   1. Upgrade to 1.26.34 w a beta having only one profile and calling brave.exe w --disable-features=ExtensionsToolbarMenu
   2. Restart Brave.
   3. 

## Actual result:
![image](https://user-images.githubusercontent.com/596147/119183408-ab03fe80-ba39-11eb-9dc6-caa08b99e23f.png)

## Expected result:
Extensions should load

## Reproduces how often: 
Easily

## Brave version (info found on brave://version)
Upgraded to Version 1.26.34 Chromium: 91.0.4472.57 (Official Build) beta (64-bit)->

## Version/Channel Information:

- Can you reproduce this issue with the current release? NO
- Can you reproduce this issue with the beta channel? YES
- Can you reproduce this issue with the dev channel? 
- Can you reproduce this issue with the nightly channel?

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? No
- Does the issue resolve itself when disabling Brave Rewards? Rewards are disabled
- Is the issue reproducible on the latest version of Chrome? No

## Miscellaneous Information:
I only had one profile in prior Brave betas so whenever brave was called in Windows 10 it was not using the -profile parameter.
If one creates a new profile aka a 2nd profile Brave will create shortcuts on your desktop. If you click on the shortcut that Brave created for the existing profile (not new profile) then the extensions load properly. However as soon as you click on the profile icon either for this profile or the newly created profile Brave crashes/closes down immediately.  There is no way of deleting profiles or managing them.

**Comment by snowbound** [comment]
Same issue occurring on 2nd Windows10 system, a laptop. However unlike the first system even launching the profile that was being used prior to this issue I still can not get the extensions to show up in the menu bar i.e. the push pin option in Extension puzzle piece does nothing.

**Comment by snowbound** [comment]
uploaded crashes 




Status: | Uploaded
-- | --
Uploaded Crash Report ID: | 83f71100-e266-5b06-0000-000000000000
Upload Time: | Friday, May 21, 2021 at 11:37:02 PM






<h3 class="capture-time" style="margin: 0.75em 0px; font-size: 12px; color: rgb(0, 0, 0); font-family: &quot;Segoe UI&quot;, Tahoma, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Crash from Friday, May 21, 2021 at 5:15:36 PM</h3>
Status: | Uploaded
-- | --
Uploaded Crash Report ID: | 7bf71100-e266-5b06-0000-000000000000
Upload Time: | Friday, May 21, 2021 at 11:36:54 PM





<h3 class="capture-time" style="margin: 0.75em 0px; font-size: 12px; color: rgb(0, 0, 0); font-family: &quot;Segoe UI&quot;, Tahoma, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><br class="Apple-interchange-newline">Crash from Friday, May 21, 2021 at 4:58:46 PM</h3>
Status: | Uploaded
-- | --
Uploaded Crash Report ID: | 73f71100-e266-5b06-0000-000000000000
Upload Time: | Friday, May 21, 2021 at 11:36:53 PM





<h3 class="capture-time" style="margin: 0.75em 0px; font-size: 12px; color: rgb(0, 0, 0); font-family: &quot;Segoe UI&quot;, Tahoma, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Crash from Friday, May 21, 2021 at 5:28:49 PM</h3>
Status: | Uploaded
-- | --
Uploaded Crash Report ID: | aaf71100-e266-5b06-0000-000000000000



<h3 class="capture-time" style="margin: 0.75em 0px; font-size: 12px; color: rgb(0, 0, 0); font-family: &quot;Segoe UI&quot;, Tahoma, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Crash from Friday, May 21, 2021 at 5:35:09 PM</h3>
Status: | Uploaded
-- | --
Uploaded Crash Report ID: | aef71100-e266-5b06-0000-000000000000
Upload Time: | Friday, May 21, 2021 at 11:37:03 PM







Status: | Uploaded
-- | --
Uploaded Crash Report ID: | b2f71100-e266-5b06-0000-000000000000
Upload Time: | Friday, May 21, 2021 at 11:37:04 PM



Upload Time: | Friday, May 21, 2021 at 11:37:03 PM





<h3 class="capture-time" style="margin: 0.75em 0px; font-size: 12px; color: rgb(0, 0, 0); font-family: &quot;Segoe UI&quot;, Tahoma, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Crash from Friday, May 21, 2021 at 5:36:58 PM</h3>
Status: | Uploaded
-- | --
Uploaded Crash Report ID: | b9f71100-e266-5b06-0000-000000000000
Upload Time: | Friday, May 21, 2021 at 11:37:05 PM





<h3 class="capture-time" style="margin: 0.75em 0px; font-size: 12px; color: rgb(0, 0, 0); font-family: &quot;Segoe UI&quot;, Tahoma, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><br class="Apple-interchange-newline">Crash from Friday, May 21, 2021 at 6:49:35 PM</h3>
Status: | Uploaded
-- | --
Uploaded Crash Report ID: | bbf71100-e266-5b06-0000-000000000000



<h3 class="capture-time" style="margin: 0.75em 0px; font-size: 12px; color: rgb(0, 0, 0); font-family: &quot;Segoe UI&quot;, Tahoma, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Crash from Friday, May 21, 2021 at 8:17:33 PM</h3>
Status: | Uploaded
-- | --
Uploaded Crash Report ID: | bef71100-e266-5b06-0000-000000000000
Upload Time: | Friday, May 21, 2021 at 11:37:05 PM




Upload Time: | Friday, May 21, 2021 at 11:37:05 PM





<h3 class="capture-time" style="margin: 0.75em 0px; font-size: 12px; color: rgb(0, 0, 0); font-family: &quot;Segoe UI&quot;, Tahoma, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Crash from Friday, May 21, 2021 at 8:19:58 PM</h3>
Status: | Uploaded
-- | --
Uploaded Crash Report ID: | bff71100-e266-5b06-0000-000000000000
Upload Time: | Friday, May 21, 2021 at 11:37:06 PM





<h3 class="capture-time" style="margin: 0.75em 0px; font-size: 12px; color: rgb(0, 0, 0); font-family: &quot;Segoe UI&quot;, Tahoma, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Crash from Friday, May 21, 2021 at 11:30:05 PM</h3>
Status: | Uploaded
-- | --
Uploaded Crash Report ID: | c4f71100-e266-5b06-0000-000000000000
Upload Time: | Friday, May 21, 2021 at 11:37:06 PM

**Comment by bsclifton** [comment]
cc: @iefremov @mkarolin

**Comment by iefremov** [comment]
Many thanks @snowbound 
this is similar to https://github.com/brave/brave-browser/issues/13822

```
[ 00 ] ui::ThemedVectorIcon::GetColor(ui::NativeTheme const *)
[ 01 ] ui::ThemedVectorIcon::GetImageSkia(ui::NativeTheme const *)
[ 02 ] views::Widget::GetNativeTheme()
[ 03 ] views::ThemedVectorIconBackground::Paint(gfx::Canvas *,views::View *)
[ 04 ] SkCanvas::clipRect(SkRect const &,SkClipOp,bool)
[ 05 ] views::View::OnPaintBackground(gfx::Canvas *)
[ 06 ] cc::RecordPaintCanvas::save()
[ 07 ] views::View::OnPaint(gfx::Canvas *)
[ 08 ] views::View::Paint(views::PaintInfo const &)
[ 09 ] void std::__1::vector<gfx::Rect,std::allocator<gfx::Rect> >::__append(unsigned __int64, const class gfx::Rect & const)
[ 10 ] SkCanvas::restore()

```
https://brave.sp.backtrace.io/p/brave/triage?time=month&filters=((callstack.functions%2Ccontains%2CThemedVectorIcon)%2C_deleted%3D0)&aggregations=((channel%2Cdistribution%2C3)%2C(ptype%2Chead)%2C(uname.sysname%2Cdistribution%2C3))&similarity=false

**Comment by snowbound** [comment]
The other machine the laptop has the problem where the Extension Puzzle piece in the toolbar the pushpins are dead in the water. the result is the end user has lost the ability to select which extensions are visible on the toolbar in Brave. On the laptop one has to click on the extension in question after clicking on the puzzle piece. Both machines are on the beta.

**Comment by snowbound** [comment]
Posted the inability on the laptop to have extensions on menu as a separate issue https://github.com/brave/brave-browser/issues/16047

**Comment by mariospr** [comment]
Hi! I was trying to reproduce this locally but I'm having trouble properly understanding what the steps to reproduce this bug are, since the description of shows several extensions correctly loaded in "Actual result" already...  but then in "Expected result" it says "extensions should load", so I'm confused.

Is it maybe the problem that some extensions don't seem to load (e.g. Better Tweetdeck)? Sorry if the question is dumb, just trying to properly understand how to trigger the crasher, because so far I could not do it.

Thanks in advance!

**Comment by snowbound** [comment]
@mariospr The screen shot for this bug report is actually from the laptop. The pushpins for each of those extensions are useless. When I click on them they do not turn red nor does the extension show up on the menubar. This extension pinning issue is separate from Brave crashing and I have created a separate bug report for the extension pinning issue at https://github.com/brave/brave-browser/issues/16047

The Brave crashing is a result of clicking on the profile icon in the menubar.

**Comment by mariospr** [comment]
@snowbound Ah, I see now, I got confused because of the title & the description apparently mixing different issues together, but now this is clearer, thanks.

Btw, now that you filed https://github.com/brave/brave-browser/issues/16047 for the extension pinning issue, what about maybe simplifying / updating this issue's title and description to better reflect what the problem here is about, exactly. Bonus points if you can also find a simplified test case and provide step-by-step instructions to reproduce the crash, so that we're sure we're talking about the same thing.

PS: I found a crash related to creating a profile but I'm not 100% certain it's the same one you're seeing, so I've just filed https://github.com/brave/brave-browser/issues/16050

**Comment by snowbound** [comment]
@mariospr The https://github.com/brave/brave-browser/issues/16047 is still happening on my laptop and is making using extensions in Brave royal pia.  I can never see the status indicator some extensions use without clicking on the puzzle piece first then looking at the extension.

**Comment by mariospr** [comment]
> @mariospr The #16047 is still happening on my laptop and is making using extensions in Brave royal pia. I can never see the status indicator some extensions use without clicking on the puzzle piece first then looking at the extension.

Hi! Just a quick comment here to say I replied about this  https://github.com/brave/brave-browser/issues/16047 (better to keep the discussion related to that problem there, thanks)

**Comment by snowbound** [comment]
Re 16034 is still happening. Brave will crash when I click on the Profile icon.




Status: | Uploaded
-- | --
Uploaded Crash Report ID: | 69830700-06d3-6d06-0000-000000000000
Upload Time: | Thursday, June 3, 2021 at 5:12:45 PM

**Comment by mariospr** [comment]
> Re 16034 is still happening. Brave will crash when I click on the Profile icon.
> 
> Status:	Uploaded
> Uploaded Crash Report ID:	69830700-06d3-6d06-0000-000000000000
> Upload Time:	Thursday, June 3, 2021 at 5:12:45 PM

@iefremov :point_up:

**Comment by snowbound** [comment]
@mariospr  A month and a half later Brave profile use on a Win10 system here is utterly useless. Brave will crash as soon as I click on the profile icon in the toolbar. Version 1.27.100 Chromium: 91.0.4472.124 (Official Build) beta (64-bit). Brave has become a wack a mole browser. Issues linger on I have to develop workarounds

**Comment by bsclifton** [comment]
Looks like `ExtensionsToolbarMenu` flag was removed - I'd definitely recommend removing that from the CLI as it won't be doing anything. If it is doing something, I'd still remove it - as It's hard to provide support for something we don't set on install.

This issue seems to describe two problems:
- extensions not loading. Was that covered by https://github.com/brave/brave-browser/issues/16047?
- Crash when clicking the profile icon (I'll edit issue to capture this as the problem)

During a regular Brave launch (no disabling features), are you still having crashes when clicking profile icon @snowbound? The profile icon shouldn't show unless you actually have more than 1 profile. If you make a new profile does it crash?

**Comment by snowbound** [comment]
@bsclifton Yes during regular launch Brave crashes every time I click on Profile Icon. If I create a  new profile I immediately see a popup saying Restore Pages Brave closed down unexpectedly. If I click on the profile icon for the new profile Brave immediately crashes. I am concerned that I will wind up being locked out of the main profile and not being able to get back into it since there is no way to select which profile I want to use.

**Comment by snowbound** [comment]
That is exactly what happened. I was unable to get back into the main profile.  I was repeatedly put back into the new profile I had created. I had to blow away the User data folder under Brave-Browser-Beta  and restore the User data folder from a recent backup.

In the User Data folder there is a Profile 1 folder. How can I tell where the main profile aka the one I use every day is storing its data.  Do I have to nuke Brave Beta off this machine entirely and recreate this profile from square one?

**Comment by bsclifton** [comment]
@snowbound visiting brave://version/ and searching for `Profile Path` will let you know the path used. For example, mine is `C:\Users\username\AppData\Local\BraveSoftware\Brave-Browser-Nightly\User Data\Default`

Can you check brave://flags and see if you have anything toggled? If you do have something manually enabled/disabled, you can set it back to `Default` to see if that helps

Worst case scenario: you should be able to do the following:
1. Move the Brave user data folder out of the way (ex: move `c:\Users\username\AppData\Local\BraveSoftware\Brave-Browser-Beta\User Data` to `c:\Users\username\AppData\Local\BraveSoftware\Brave-Browser-Beta\User Data2`)
2. Launch Brave and verify that you don't get any crashes out of the box. You can try creating a new profile and switching between them; then delete the new profile after you test
3. Quit Brave
4. Perform some "rocket surgery"
    - basically copy the files from the old folder, specifically ONLY copying the files in the profile directory (ex: `c:\Users\username\AppData\Local\BraveSoftware\Brave-Browser-Beta\User Data2\Default` or `c:\Users\username\AppData\Local\BraveSoftware\Brave-Browser-Beta\User Data2\Person 1`, whatever the name is). 
    - Copy these files over into the NEW folder which will be named `c:\Users\username\AppData\Local\BraveSoftware\Brave-Browser-Beta\User Data\Default`

**Comment by snowbound** [comment]
@bsclifton  I do not think the "rocket surgery" will be needed and that is a good thing since it is a Monday here and Mondays have a habit of going south here quickly ;)

What I found when I went into the flags was something called "New Profile Picker" was enabled. Why this was enabled is beyond me since I have only ever had one profile set up in Brave. Once I set that to Default restarted Brave I was able to click on the Profile button, go into the menu and delete the test profiles I had created during this issue. I was also able to create new profiles and delete them. 

With the single profile, I now have no profile icon on the toolbar as it should be. Thanks for the help. Hopefully, this will stay fixed but time will tell.

**Comment by bsclifton** [comment]
@snowbound very cool! Super glad you were able to get sorted - and you helped narrow down the problem 😄 I updated the issue title and gave it a priority

cc: @brave/patch-reviewers - crash is happening with brave://flags/#enable-new-profile-picker enabled

**Comment by iefremov** [comment]
stale, also we don't guarantee stable work when any of the upstream flags are toggled
